typescript-carousel/main.ts

```diff
@@ -0,0 +1,64 @@
+const $images = document.querySelectorAll('img');
+const $dots = document.querySelectorAll('.fa-circle');
+const $rightArrow = document.querySelector('.fa-arrow-right');
+const $leftArrow = document.querySelector('.fa-arrow-left');
+const $iconCol = document.querySelector('.icon-col');
+if (!$rightArrow || !$leftArrow || !$iconCol)
+  throw new Error('$rightArrow, $leftArrow, $iconCol query failed');
+let position = 0;
+
+let intervalID = setInterval(upSwitchIcon, 3000);
+
+function upSwitchIcon(): void {
+  if (position === 4) {
```

Comment:
Only thing I would change is by using the `length` of your `$images`. That way you can add more images to your HTML and it will still work, and you wouldn't need to update the position everytime